docs/how-to/hip_runtime_api/memory_management/unified_memory.rst

```diff
@@ -0,0 +1,737 @@
+
+                // Wait for GPU to finish before accessing on host.
+                HIP_CHECK(hipDeviceSynchronize());
+
+                // Prints the result.
+                std::cout << *a << " + " << *b << " = " << *c << std::endl;
+
+                // Cleanup allocated memory.
+                delete[] a;
+                delete[] b;
+                delete[] c;
+
+                return 0;
+            }
+
+    .. tab-item:: Explicit Memory Management
+
+        .. code-block:: cpp
+            :emphasize-lines: 27-34, 39-40
+
+            #include <hip/hip_runtime.h>
+            #include <iostream>
+
+            #define HIP_CHECK(expression)              \
+            {                                          \
+                const hipError_t err = expression;     \
+                if(err != hipSuccess){                 \
+                    std::cerr << "HIP error: "         \
+                        << hipGetErrorString(err)      \
+                        << " at " << __LINE__ << "\n"; \
+                }                                      \
+            }
+
+            // Addition of two values.
+            __global__ void add(int *a, int *b, int *c) {
+                *c = *a + *b;
+            }
+
+            int main() {
+                int a, b, c;
+                int *d_a, *d_b, *d_c;
+
+                // Setup input values.
+                a = 1;
+                b = 2;
+
+                // Allocate device copies of a, b and c.
+                HIP_CHECK(hipMalloc(&d_a, sizeof(*d_a)));
+                HIP_CHECK(hipMalloc(&d_b, sizeof(*d_b)));
+                HIP_CHECK(hipMalloc(&d_c, sizeof(*d_c)));
+
+                // Copy input values to device.
+                HIP_CHECK(hipMemcpy(d_a, &a, sizeof(*d_a), hipMemcpyHostToDevice));
+                HIP_CHECK(hipMemcpy(d_b, &b, sizeof(*d_b), hipMemcpyHostToDevice));
+
+                // Launch add() kernel on GPU.
+                hipLaunchKernelGGL(add, dim3(1), dim3(1), 0, 0, d_a, d_b, d_c);
+
+                // Copy the result back to the host.
+                HIP_CHECK(hipMemcpy(&c, d_c, sizeof(*d_c), hipMemcpyDeviceToHost));
+
+                // Cleanup allocated memory.
+                HIP_CHECK(hipFree(d_a));
+                HIP_CHECK(hipFree(d_b));
+                HIP_CHECK(hipFree(d_c));
+
+                // Prints the result.
+                std::cout << a << " + " << b << " = " << c << std::endl;
+
+                return 0;
+            }
+
+.. _using unified memory:
+
+Using unified memory
+================================================================================
+
+Unified memory can simplify the complexities of memory management in GPU
+computing, by not requiring explicit copies between the host and the devices. It
+can be particularly useful in use cases with sparse memory accesses from both
+the CPU and the GPU, as not the whole memory region needs to be transferred to
```

Comment:
```suggestion
the CPU and the GPU, as the whole memory region does not need to be transferred to
```

Reply:
There is a logical difference between "not the whole memory region needs to be transferred" and "the whole memory region does not need to be transferred"

Reply:
What is the meaning of the sentence?

Reply:
Changed the sentence to make it easier to understand.